src/core/utilities.cpp

```diff
@@ -564,24 +559,19 @@ bool ParseUntilElement(QXmlStreamReader* reader, const QString& name) {
 }
 
 QDateTime ParseRFC822DateTime(const QString& text) {
-  // This sucks but we need it because some podcasts don't quite follow the
-  // spec properly - they might have 1-digit hour numbers for example.
-  QDateTime ret;
-  QRegExp re(
-      "([a-zA-Z]{3}),? (\\d{1,2}) ([a-zA-Z]{3}) (\\d{4}) "
-      "(\\d{1,2}):(\\d{1,2}):(\\d{1,2})");
-  if (re.indexIn(text) != -1) {
-    ret = QDateTime(
-        QDate::fromString(QString("%1 %2 %3 %4")
-                              .arg(re.cap(1), re.cap(3), re.cap(2), re.cap(4)),
-                          Qt::TextDate),
-        QTime(re.cap(5).toInt(), re.cap(6).toInt(), re.cap(7).toInt()));
+  QRegExp regexp("(\\d{1,2}) (\\w{3,12}) (\\d+) (\\d{1,2}):(\\d{1,2}):(\\d{1,2})");
+  if (regexp.indexIn(text) == -1) {
+    return QDateTime();
   }
-  if (ret.isValid()) return ret;
-  // Because http://feeds.feedburner.com/reasonabledoubts/Msxh?format=xml
-  QRegExp re2(
-      "(\\d{1,2}) ([a-zA-Z]{3}) (\\d{4}) "
-      "(\\d{1,2}):(\\d{1,2}):(\\d{1,2})");
+
+  enum match_names {
```

Comment:
Ugh, it's a shame QRegExp doesn't support named groups.
To make the style a little nicer can you make this an enum class, rename it MatchNames and rename the values to be in UPPER_CASE.